js/mcqView.js

```diff
@@ -0,0 +1,186 @@
+                }
+                $itemLabel.addClass('selected');
+                $itemLabel.a11y_selected(true);
+            } else {
+                selectedItems.splice(_.indexOf(selectedItems, item), 1);
+                $itemLabel.removeClass('selected');
+                $itemLabel.a11y_selected(false);
+            }
+            $itemInput.prop('checked', selected);
+            item._isSelected = selected;
+            this.model.set('_selectedItems', selectedItems);
+        },
+
+        // Blank method to add functionality for when the user cannot submit
+        // Could be used for a popup or explanation dialog/hint
+        onCannotSubmit: function() {},
+
+        // This is important and should give the user feedback on how they answered the question
+        // Normally done through ticks and crosses by adding classes
+        showMarking: function() {
+            _.each(this.model.get('_items'), function(item, i) {
+                var $item = this.$('.component-item').eq(i);
+                $item.removeClass('correct incorrect').addClass(item._isCorrect ? 'correct' : 'incorrect');
+            }, this);
+        },
+
+        // Used by the question view to reset the look and feel of the component.
+        resetQuestion: function() {
+            this.deselectAllItems();
+            this.resetItems();
+        },
+
+        deselectAllItems: function() {
+            this.$el.a11y_selected(false);
+            this.model.deselectAllItems();
+        },
+
+        resetItems: function() {
+            this.$('.component-item label').removeClass('selected');
+            this.$('.component-item').removeClass('correct incorrect');
+            this.$('input').prop('checked', false);
+            this.model.resetItems();
+        },
+
+        showCorrectAnswer: function() {
+            _.each(this.model.get('_items'), function(item, index) {
+                this.setOptionSelected(index, item._shouldBeSelected);
+            }, this);
+        },
+
+        setOptionSelected:function(index, selected) {
+            var $itemLabel = this.$('label').eq(index);
+            var $itemInput = this.$('input').eq(index);
+            if (selected) {
+                $itemLabel.addClass('selected');
+                $itemInput.prop('checked', true);
+            } else {
+                $itemLabel.removeClass('selected');
+                $itemInput.prop('checked', false);
+            }
+        },
+
+        hideCorrectAnswer: function() {
+            _.each(this.model.get('_items'), function(item, index) {
+                this.setOptionSelected(index, this.model.get('_userAnswer')[item._index]);
+            }, this);
+        },
+
+        /**
+        * used by adapt-contrib-spoor to get the user's answers in the format required by the cmi.interactions.n.student_response data field
+        * returns the user's answers as a string in the format "1,5,2"
+        */
+        getResponse:function() {
```

Comment:
small point but feels like `getResponse` and `getResponseType` really belong to the model - but of course this would require more specific models than we currently have